packages/core/src/services/zoom.ts

```diff
@@ -64,4 +70,59 @@ export class Zoom extends Service {
 			extent(zoomBarData, (d: any) => d[domainIdentifier])
 		);
 	}
+
+	isZoomBarEnabled() {
+		// @todo - need to update this if zoom bar in other position (bottom, left, right) is supported
+		// check configuration
+		if (
+			!Tools.getProperty(
+				this.model.getOptions(),
+				"zoomBar",
+				"top",
+				"enabled"
+			)
+		) {
+			return false;
+		}
+
+		// CartesianScales service is only available in axis charts
+		if (!this.services.cartesianScales) {
+			return false;
+		}
+
+		// @todo - Zoom Bar only supports main axis at BOTTOM axis and time scale for now
+		this.services.cartesianScales.findDomainAndRangeAxes(); // need to do this before getMainXAxisPosition()
+		const mainXAxisPosition = this.services.cartesianScales.getMainXAxisPosition();
+		const mainXScaleType = Tools.getProperty(
+			this.model.getOptions(),
+			"axes",
+			mainXAxisPosition,
+			"scaleType"
+		);
+
+		return (
+			mainXAxisPosition === AxisPositions.BOTTOM &&
+			mainXScaleType === ScaleTypes.TIME
+		);
+	}
+
+	isToolbarEnabled() {
+		return (
+			this.isZoomBarEnabled() &&
```

Comment:
this shouldn't be the case, and not sure if `isToolbarEnabled` can live in the zoom service

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/50a9137867775ff491405b4b22971d9968682fa8.